code/modules/vehicle/bicycle.dm

```diff
@@ -0,0 +1,62 @@
+/obj/vehicle/bike
+	name = "bicycle"
+	desc = "Two wheels of FURY!"
+	icon = 'icons/obj/vehicles.dmi'
+	icon_state = "bicycle"
+	vehicle_move_delay = 1
+	var/mutable_appearance/bicycle_overlay
+
+/obj/vehicle/bike/Initialize(mapload)
+	. = ..()
+	bicycle_overlay = mutable_appearance(icon, "bicycle_overlay", ABOVE_MOB_LAYER)
+
+/obj/vehicle/bike/relaymove(mob/user, direction)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/driver = user
+	var/obj/item/organ/external/l_hand = driver.get_organ("l_hand")
+	var/obj/item/organ/external/r_hand = driver.get_organ("r_hand")
+	if(!l_hand && !r_hand)
+		vehicle_move_delay += 0.5 // I can ride my bike with no handlebars... (but it's slower)
+	for(var/organ_name in list("l_leg", "r_leg", "l_foot", "r_foot"))
+		var/obj/item/organ/external/E = driver.get_organ(organ_name)
+		if(!E)
+			return // Bikes need both feet/legs to work. missing even one makes it so you can't ride the bike
+		if(E.status & ORGAN_SPLINTED)
+			vehicle_move_delay += 0.5
+		else if(E.status & ORGAN_BROKEN)
+			vehicle_move_delay += 1.5
+
+/obj/vehicle/bike/post_buckle_mob(mob/living/M)
+	. = ..()
+	var/datum/action/bicycle_bell/bell_action = new(src)
+	if(has_buckled_mobs())
+		bell_action.Grant(M)
+	else
+		bell_action.Remove(M)
```

Comment:
When would we have this? Wouldn't this be handled by `post_unbuckle_mob`?

Reply:
Removed, it was truly not necessary from investigating it now.